amqp-impl/src/main/java/io/streamnative/pulsar/handlers/amqp/ExchangeMessageRouter.java

```diff
@@ -0,0 +1,474 @@
+
+            void addBinding(String routingKey) {
+                if (bindingKeys.add(routingKey)) {
+                    topicParser = new TopicParser();
+                    topicParser.addBinding(routingKey, null);
+                }
+            }
+
+            void unbind(String routingKey) {
+                bindingKeys.remove(routingKey);
+                topicParser = new TopicParser();
+                for (String bindingKey : bindingKeys) {
+                    topicParser.addBinding(bindingKey, null);
+                }
+            }
+
+        }
+
+        @Override
+        public synchronized void addBinding(String des, String desType, String routingKey,
+                                            Map<String, Object> arguments) {
+            destinationMap.computeIfAbsent(new Destination(des, desType), k -> new TopicRoutingKeyParser())
+                    .addBinding(routingKey);
+        }
+
+        @Override
+        public synchronized void removeBinding(String des, String desType, String routingKey,
+                                               Map<String, Object> arguments) {
+            destinationMap.computeIfPresent(new Destination(des, desType), (k, v) -> {
+                v.unbind(routingKey);
+                if (v.bindingKeys.isEmpty()) {
+                    return null;
+                }
+                return v;
+            });
+        }
+
+        @Override
+        Set<Destination> getDestinations(String routingKey, Map<String, Object> headers) {
+            Set<Destination> destinations = new HashSet<>();
+            for (Map.Entry<Destination, TopicRoutingKeyParser> entry : destinationMap.entrySet()) {
+                if (!entry.getValue().topicParser.parse(routingKey).isEmpty()) {
+                    destinations.add(entry.getKey());
+                }
+            }
+            return destinations;
+        }
+    }
+
+    static class HeadersExchangeMessageRouter extends ExchangeMessageRouter {
+
+        private final Map<Destination, HeadersMessageRouter> messageRouterMap;
+
+        public HeadersExchangeMessageRouter(PersistentExchange exchange, ExecutorService routeExecutor) {
+            super(exchange, routeExecutor);
+            messageRouterMap = new ConcurrentHashMap<>();
+        }
+
+        @Override
+        public synchronized void addBinding(String des, String desType, String routingKey,
+                                            Map<String, Object> arguments) {
+            messageRouterMap.computeIfAbsent(new Destination(des, desType), k -> new HeadersMessageRouter())
+                    .getArguments().putAll(arguments);
+        }
+
+        @Override
+        public synchronized void removeBinding(String des, String desType, String routingKey,
+                                               Map<String, Object> arguments) {
+            messageRouterMap.computeIfPresent(new Destination(des, desType), (k, v) -> {
+                v.getArguments().putAll(arguments);
```

Comment:
need to delete

Reply:
The `arguments` is null while removing binding. I change this to remove the destination directly.